### ISSUE TITLE
No glibcLocalesUtf8 in nix-pkgs revisions corresponding to R <= 4.2.0

### ISSUE THREAD

**Issue by motorlearner** [b-rodrigues/rix#95]
When requesting R version 4.2.0 (or lower as far as I can tell), I get

```
...
error: attribute 'glibcLocalesUtf8' missing

at /path/to/project/default.nix:20:43:

           19| };
           20|    system_packages = builtins.attrValues {
             |                                           ^
           21|   inherit (pkgs) R glibcLocalesUtf8 ;
...
```

I am using rix version 0.5.0
```r
packageVersion("rix")
# [1] ‘0.5.0’
```

Edit: 
as a workaround, I am currently using a fixed (newer) revision to load glibcLocalesUtf8

**Comment by philipp-baumann** [comment]
Thanks a lot @motorlearner for reporting. Can confirm this. I wasn't aware that the `glibcLocalesUtf8` was added to nixpkgs relatively recently. Importing entire `glibcLocales` seems a bit heavy, this is why we choose `glibcLocalsUtf8`, but since this is not sane for earlier nixpkgs revisions this is for sure not a sane default.

**Comment by philipp-baumann** [comment]
Importing all and running `nix-shell --pure .` like this did the trick on my Rocky test server.

```sh
# This file was generated by the {rix} R package v0.5.0 on 2024-01-15
# with following call:
# >rix(r_ver = "f597e7e9fcf37d8ed14a12835ede0a7d362314bd")
# It uses nixpkgs' revision f597e7e9fcf37d8ed14a12835ede0a7d362314bd for reproducibility purposes
# which will install R version 4.2.0
# Report any issues to https://github.com/b-rodrigues/rix
let
  pkgs = import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/f597e7e9fcf37d8ed14a12835ede0a7d362314bd.tar.gz") {};
  system_packages = builtins.attrValues {
    inherit (pkgs) R glibcLocales;
  };
 in
  pkgs.mkShell {
    LOCALE_ARCHIVE = if pkgs.system == "x86_64-linux" then  "${pkgs.glibcLocales}/lib/locale/locale-archive" else "";
    LANG = "en_US.UTF-8";
    LC_ALL = "en_US.UTF-8";
    LC_TIME = "en_US.UTF-8";
    LC_MONETARY = "en_US.UTF-8";
    LC_PAPER = "en_US.UTF-8";
    LC_MEASUREMENT = "en_US.UTF-8";

    buildInputs = [    system_packages  ];
      shellHook = "R --vanilla";
  }
```

<img width="911" alt="Screenshot 2024-01-15 at 20 09 05" src="https://github.com/b-rodrigues/rix/assets/21625034/4daae8d1-53a3-4a2d-9e0c-8e463ad6b065">

The other maybe a bit cheap and not elegant solution is just not setting `LOCALE_ARCHIVE` and locale settings. Is not so nice in R because it gives a warning at startup.

**Comment by b-rodrigues** [comment]
How much space is the whole glibclocales?

@motorlearner could you share the default.nix you used with the workaround? we could perhaps always do that...

**Comment by philipp-baumann** [comment]
> How much space is the whole glibclocales?
> 
> @motorlearner could you share the default.nix you used with the workaround? we could perhaps always do that...

right now have `nix path-info` cmd not activated, but according to this its a bit more than 200MB. https://github.com/NixOS/nixpkgs/issues/85823#issuecomment-674224421

**Comment by b-rodrigues** [comment]
200mb, honestly I'd say we just install the whole thing for older versions of R

**Comment by philipp-baumann** [comment]
> 200mb, honestly I'd say we just install the whole thing for older versions of R

yeah, let's do that. Rather not mess with reproducibility.

**Comment by philipp-baumann** [comment]
> > 200mb, honestly I'd say we just install the whole thing for older versions of R
> 
> yeah, let's do that. Rather not mess with reproducibility.

just wanna make sure it works on MacOS first.

**Comment by b-rodrigues** [comment]
> Rather not mess with reproducibility.

Amen!

**Comment by philipp-baumann** [comment]
<img width="913" alt="Screenshot 2024-01-15 at 21 01 23" src="https://github.com/b-rodrigues/rix/assets/21625034/d7694bec-e34c-4b5c-afab-2c3884d7e511">
does work.

**Comment by philipp-baumann** [comment]
probably good to update also `inst/extdata/default.nix` after the merge, and then bump patch.

**Comment by motorlearner** [comment]
Works for me on MacOS 14.2.1.

> ```r
> remotes::install_github("b-rodrigues/rix", ref = "iss95-patch-glibclocales")
> library(rix) # else r_nix_revs not found
> rix::rix(r_ver = "4.2.0")
> ```
> 
> ```shell
> nix-shell --pure .
> ```

**Comment by philipp-baumann** [comment]
> Works for me on MacOS 14.2.1.
> 
> > ```r
> > remotes::install_github("b-rodrigues/rix", ref = "iss95-patch-glibclocales")
> > library(rix) # else r_nix_revs not found
> > rix::rix(r_ver = "4.2.0")
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```shell
> > nix-shell --pure .
> > ```

great, also on MacOS 14.2.1. Can also confirm it works on Ubuntu 22.04.3 LTS. For the record, we can go back until R 3.4.0; we can not go to nixpkgs revisions that correspond to earlier R versions, since we get `"error: attribute 'mkShell'" missing`.

**Comment by philipp-baumann** [comment]
fixed with tagged patch version rix 0.5.1 https://github.com/b-rodrigues/rix/commit/28706de31f814eab5ba603d022befc0261a02761

**Comment by b-rodrigues** [comment]
Late to the party, but can confirm it works on Ubuntu 23.10 for ARM (rpi 5)
